src/core/layout/qgslayoutitemscalebar.cpp

```diff
@@ -570,10 +605,31 @@ double QgsLayoutItemScaleBar::mapWidth() const
     da.setEllipsoid( mLayout->project()->ellipsoid() );
 
     const Qgis::DistanceUnit units = da.lengthUnits();
-    double measure = da.measureLine( QgsPointXY( mapExtent.xMinimum(), mapExtent.yMinimum() ),
-                                     QgsPointXY( mapExtent.xMaximum(), mapExtent.yMinimum() ) );
-    measure /= QgsUnitTypes::fromUnitToUnitFactor( mSettings.units(), units );
-    return measure;
+
+    // measure across top, center, and bottom of map, and return average of these calculated lengths
+    int validMeasureCount = 0;
+    double sumValidMeasures = 0;
+    for ( const double y : { mapExtent.yMinimum(), mapExtent.yMaximum(), ( mapExtent.yMaximum() - mapExtent.yMinimum() ) * 0.5 } )
```

Comment:
```suggestion
    for ( const double y : { mapExtent.yMinimum(), mapExtent.yMaximum(), ( mapExtent.yMaximum() + mapExtent.yMinimum() ) * 0.5 } )
```
Shouldn't it be `( mapExtent.yMaximum() + mapExtent.yMinimum() ) * 0.5`?